File: src/main/c/server/ghost.c
#include "ghosts.h"

void *ghosts_move(void *arg) {

	struct game *g = (struct game *) arg;

	while(g->remaining_ghosts > 0) {
		for(int i=0;i<MAX_GHOST_NUMBER;i++) {
			if(g->ghosts[i].x != -1) {
				while(1) {
					int tmp_x = rand() % g->labyrinth->width;
					int tmp_y = rand() % g->labyrinth->height;
					if(g->labyrinth->cells[tmp_y][tmp_x] != 1) {
						g->ghosts[i].x = tmp_x;
						g->ghosts[i].y = tmp_y;
						break;
					}
				}
			}	
		}
		usleep(GHOST_MOVE_FREQUENCY);	
		puts("ghosts moved");
		//TODO: send ghost positions 
	}

	return NULL;
}

File: src/main/java/game/Ghost.java
package main.java.game;

public class Ghost {
    private final int x;
    private final int y;

    public Ghost(int x, int y) {
        this.x = x;
        this.y = y;
    }

    public int getPosX() {
        return x;
    }
    public int getPosY() {
        return y;
    }
}
